tests/ga/test_signature_validation_sudo.py

```diff
@@ -77,3 +93,84 @@ def test_should_validate_signature_for_package_signed_with_leaf_root_cert(self):
+        with patch("azurelinuxagent.ga.signature_validation_util.CGroupConfigurator.get_instance") as mock_get_instance:
+            mock_instance = mock_get_instance.return_value
+            mock_instance.enabled.return_value = True
+            
+            def mock_run_command_with_error(cmd, *args, **kwargs):
+                # Simulate OpenSSL failure (unit name in stderr means it's not a systemd failure)
+                if 'openssl' in ' '.join(cmd):
+                    raise shellutil.CommandError(command=cmd, return_code=1, stdout='',
+                                                 stderr='Running as unit: {0}\nVerification failure'.format(EXT_SIGNATURE_VALIDATION_CGROUPS_UNIT))
+                return shellutil.run_command(cmd, *args, **kwargs)
+            
+            with patch("azurelinuxagent.ga.signature_validation_util.run_command", side_effect=mock_run_command_with_error):
+                with self.assertRaises(SignatureValidationError):
+                    validate_signature(self.vm_access_zip_path, self.vm_access_signature, self.package_name_and_version)
+
+    def test_validate_signature_should_retry_on_systemd_error(self):
+        with patch("azurelinuxagent.ga.signature_validation_util.CGroupConfigurator.get_instance") as mock_get_instance:
+            mock_instance = mock_get_instance.return_value
+            mock_instance.enabled.return_value = True
+
+            original_run_command = shellutil.run_command
+            calls = []
+
+            def run_command_with_systemd_failure(cmd, *args, **kwargs):
+                calls.append(cmd)
+                # Fail if command contains systemd-run (simulating systemd not available)
+                if 'systemd-run' in ' '.join(cmd):
+                    raise shellutil.CommandError(command=cmd, return_code=1, stdout='',
+                                                 stderr='Unit {0} not found.'.format(EXT_SIGNATURE_VALIDATION_CGROUPS_UNIT))
+                return original_run_command(cmd, *args, **kwargs)
+
+            with patch("azurelinuxagent.ga.signature_validation_util.run_command", side_effect=run_command_with_systemd_failure):
+                validate_signature(self.vm_access_zip_path, self.vm_access_signature, self.package_name_and_version)
+
+                self.assertEqual(2, len(calls))
+                # First command should be invoked via systemd-run
+                self.assertIn('systemd-run', ' '.join(calls[0]))
+                # Second command should be a direct openssl call (no systemd-run)
+                self.assertNotIn('systemd-run', ' '.join(calls[1]))
+
+                # Verify that cgroups were disabled
+                self.assertEqual(1, mock_instance.disable.call_count, "disable() should have been called exactly once")
+                reason = mock_instance.disable.call_args[1]['reason']
+                self.assertIn("'systemd-run' invocation failed for signature validation", reason)
```

Comment:
in several checks you do "foo in formatted_command" (or similar), could you be more strict and so "startswith" or command[0] == 'foo' (or similar)